ricecooker/classes/nodes.py

```diff
@@ -798,3 +798,89 @@ def truncate_fields(self):
+            thumbnail (str): local path or url to thumbnail image (optional)
+            extra_fields (dict): any additional data needed for node (optional)
+            domain_ns (str): who is providing the content (e.g. learningequality.org) (optional)
+    """
+    kind = content_kinds.SLIDESHOW
+
+    def __init__(self, source_id, title, license, slideshow_data=None, **kwargs):
+        if slideshow_data:
+            extra_fields = {'slideshow_data': slideshow_data}
+        else:
+            extra_fields = {'slideshow_data':[]}
+        # THe Node base class' __init__ method has:
+        #       for f in files or []:
+        #           self.add_file(f)
+        super(SlideshowNode, self).__init__(source_id, title, license, extra_fields=extra_fields, **kwargs)
+
+    def add_file(self, file_to_add):
+        """
+        Add a the SlideImageFile to the node's files and append slideshow metadata
+        to extra_fields['slideshow_data'] (list).
+        Args: file (SlideshowNode or ThumbnailFile): file model to add to node
+        Returns: None
+        """
+        from .files import File, SlideImageFile
+        assert isinstance(file_to_add, File), "Files being added must be instances of a subclass of File class"
+        # TODO asserr SlideshowNode OR ThumbnailFile
+
+        if file_to_add not in self.files:
+            filename = file_to_add.get_filename()
+            if filename:
+                checksum, ext = filename.split('.')  # <md5sum(contents)>.[png|jpg|jpeg]
+            else:
+                raise ValueError('filename not available')
+
+            #
+            # Appending to extra_fields is only necessary for SlideImageFile instances
+            if isinstance(file_to_add, SlideImageFile):
+                #
+                # Find the idx of sort_order.next()
+                slideshow_image_files = [f for f in self.files if isinstance(f,SlideImageFile)]
+                idx = len(slideshow_image_files)  # next available index, assuming added in desired order
+                #
+                # Add slideshow data to extra_fields['slideshow_data'] (aka manifest)
+                slideshow_data = self.extra_fields['slideshow_data']
+                slideshow_data.append(
+                    {
+                        'caption': file_to_add.caption,
+                        'descriptive_text': file_to_add.descriptive_text,
+                        'sort_order': idx,
+                        'checksum': checksum,
+                        'extension': ext
+                    }
+                )
+                self.extra_fields['slideshow_data'] = slideshow_data
+
+            #
+            # Add node->file link
+            file_to_add.node = self
+            self.files.append(file_to_add)
+
+    def validate(self):
+        from .files import SlideImageFile, ThumbnailFile
+        try:
+            assert any(self.files), "Assumption Failed: Slideshow does not have any slideshow image files."
```

Comment:
I think you want this to be something like
```
assert [f for f in self.files if isinstance(f, SlideImageFile)], \
    "Assumption Failed: SlideshowNode must have at least one SlideImageFile file."
```

here we're using the fact that empty list `[]` is falsy and non-empty list is truty

Reply:
The lines below that take care of that logic, right? I also included the assertion that it is SlideImageFile OR ThumbnailFile in the lines below using a for loop.